includes/class-micropub-media.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+add_action( 'plugins_loaded', array( 'Micropub_Media', 'init' ) );
+
+/**
+ * Micropub Media Class
+ */
+class Micropub_Media {
+
+	protected static $scopes                 = array();
+	protected static $micropub_auth_response = array();
+
+	/**
+	 * Initialize the plugin.
+	 */
+	public static function init() {
+		$cls = get_called_class();
+
+		// register endpoint
+		add_action( 'rest_api_init', array( $cls, 'register_route' ) );
+
+	}
+
+	public static function register_route() {
+		$cls = get_called_class();
+		register_rest_route(
+			'micropub/1.0', '/media', array(
+				array(
+					'methods'  => WP_REST_Server::CREATABLE,
+					'callback' => array( $cls, 'upload_handler' ),
+				),
+			)
+		);
+	}
+
+	// Based on WP_REST_Attachments_Controller function of the same name
+
+	public static function upload_from_file( $files, $name ) {
+
+		// Pass off to WP to handle the actual upload.
+		$overrides = array(
+			'test_form' => false,
+		);
+
+		// Bypasses is_uploaded_file() when running unit tests.
+		if ( defined( 'DIR_TESTDATA' ) && DIR_TESTDATA ) {
+			$overrides['action'] = 'wp_handle_mock_upload';
+		}
+
+		/** Include admin functions to get access to wp_handle_upload() */
+		require_once ABSPATH . 'wp-admin/includes/admin.php';
+
+		$file = wp_handle_upload( $files[ $name ], $overrides );
+
+		if ( isset( $file['error'] ) ) {
+			return new WP_Micropub_Error( 'invalid_request', $file['error'], 500 );
+		}
+
+		return $file;
+	}
+
+	/**
+	* Check scope
+	*
+	* @param array $scope
+	*
+	* @return boolean
+	**/
+	protected static function check_scopes( $scopes ) {
```

Comment:
minor nit: based on this method name, it surprised me that this only checks for intersection and not superset. consider either renaming to `has_any_scope`, or probably better, just inline it on line 85 since it's only called there.

Reply:
ping

Reply:
I fixed this one

Reply:
ah, you did! i missed it since this method is still here, just now empty and unused. i expect you'll want to remove it. 😁